src/main/java/net/ltxprogrammer/changed/latexvariant/CheckCondition.java

```diff
@@ -0,0 +1,58 @@
+package net.ltxprogrammer.changed.latexvariant;
+
+import io.github.edwinmindcraft.origins.api.OriginsAPI;
+import io.github.edwinmindcraft.origins.api.capabilities.IOriginContainer;
+import io.github.edwinmindcraft.origins.api.origin.Origin;
+import io.github.edwinmindcraft.origins.api.origin.OriginLayer;
+import net.ltxprogrammer.changed.Changed;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerPlayer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Objects;
+
+
+public class CheckCondition {
+    private final ServerPlayer player;
+    private static boolean conditionValue = false;
+    private static VariantType variantType = VariantType.DEFAULT;
+    public static final Logger LOGGER = LogManager.getLogger(Changed.class);
+
+    public CheckCondition(ServerPlayer player) {
```

Comment:
All of the logic in this class is only performed once, therefor constructing this class for a single boolean value and then throwing away the class is pointless.